packages/next/server/node-web-streams-helper.ts

```diff
@@ -257,16 +257,56 @@ export function createSuffixStream(
   })
 }
 
+export function createRootLayoutValidatorStream(): TransformStream<
+  Uint8Array,
+  Uint8Array
+> {
+  let foundHtml = false
+  let foundHead = false
+  let foundBody = false
+
+  return new TransformStream({
+    async transform(chunk, controller) {
+      const content = decodeText(chunk)
```

Comment:
Can  there be a fast path when html, head, and body have been found already where it doesn't have to decode as part of this transform? 🤔

Reply:
Sounds good 👍